BootloaderCommonPkg/Include/Library/ContainerLib.h

```diff
@@ -23,6 +23,12 @@
 #define AUTH_TYPE_SIG_RSA2048_SHA256  3
 #define AUTH_TYPE_SIG_RSA3072_SHA384  4
 
+// Flags for CONTAINER_HDR
+#define CONTAINER_HDR_FLAG_MONO_SIGNING     BIT0
+
+// Attributes for COMPONENT_ENTRY
+#define COMPONENT_ENTRY_ATTR_RESERVED       BIT7
+
```

Comment:
Could we add more comments? not sure why COMPONENT_ENTRY_ATTR_RESERVED is required.
Container has subregion, pod has't subregion, only has files, both using container header, how we know it is container or pod image, by CONTAINER_HDR_FLAG_MONO_SIGNING?

Reply:
CONTAINER and POD are interchangeable. CONTAINER or POD has got the region/components inside them.

Reply:
has sub-region in the container need verify header firstly, and has files in the container doesn't need verify the header separately since it will be verified with its payload. so I guess you are using CONTAINER_HDR_FLAG_MONO_SIGNING to diff them.

Reply:
what COMPONENT_ENTRY_ATTR_RESERVED is used for?

Reply:
COMPONENT_ENTRY_ATTR_RESERVED is used to set the _SG_ (last entry in the container) component as reserved, so that cant be updated or touched.

Reply:
What the last entry used for (to get authtype for container header?). could we define a entry signature, e.g. "_HDR", instead of adding a new attribute field.

Reply:
The last entry holds the AUTH data for the whole POD image.  Instead of adding the auth data at the end, we reuse a normal component entry to provide the auth info. In this way, the flow is the same and the auth algo can be determined use the same way as normal component.

Reply:
there are AuthType in container header, and AuthType in entry header.  it is a little confusing to use the AuthType in the entry header for whole POD image.  do we use AuthType in container header in POD image case?

Reply:
Yes, we still use the container header AuthType to verify the header only.   Header authentication only control header, not the POID image contents.   I think using a reserved entry to do the POD signing is a cleaner solution.  In this way for POD image, it follows the same flow as normal container verification.  There is no difference at all.